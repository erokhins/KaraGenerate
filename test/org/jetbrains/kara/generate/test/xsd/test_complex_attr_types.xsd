<?xml version='1.0' encoding='utf-8' ?>
<xsd:schema xmlns='test' xmlns:xsd='http://www.w3.org/2001/XMLSchema' version="1.0" targetNamespace='test'>

    <xsd:element name="complex">
        <xsd:complexType>
            <xsd:attribute name="attrStrEnumType1">
                <xsd:simpleType>
                    <xsd:union>
                        <xsd:simpleType>
                            <xsd:restriction base="xsd:NMTOKEN">
                                <xsd:enumeration value="value1"/>
                                <xsd:enumeration value="value2"/>
                            </xsd:restriction>
                        </xsd:simpleType>
                        <xsd:simpleType>
                            <xsd:restriction base="xsd:string"/>
                        </xsd:simpleType>
                    </xsd:union>
                </xsd:simpleType>
            </xsd:attribute>

            <xsd:attribute name="attrStrEnumType2">
                <xsd:simpleType>
                    <xsd:union>
                        <xsd:simpleType>
                            <xsd:restriction base="xsd:string">
                                <xsd:enumeration value="value-1"/>
                                <xsd:enumeration value="value-2"/>
                            </xsd:restriction>
                        </xsd:simpleType>
                        <xsd:simpleType>
                            <xsd:restriction base="xsd:string"/>
                        </xsd:simpleType>
                    </xsd:union>
                </xsd:simpleType>
            </xsd:attribute>

            <xsd:attribute name="attrUnionEnumType">
                <xsd:simpleType>
                    <xsd:union>
                        <xsd:simpleType>
                            <xsd:restriction base="xsd:NMTOKEN">
                                <xsd:enumeration value="value1"/>
                                <xsd:enumeration value="value2"/>
                            </xsd:restriction>
                        </xsd:simpleType>
                    </xsd:union>
                </xsd:simpleType>
            </xsd:attribute>

            <xsd:attribute name="attrRestrictionEnumType">
                <xsd:simpleType>
                    <xsd:restriction base="xsd:NMTOKEN">
                        <xsd:enumeration value="value1"/>
                        <xsd:enumeration value="value2"/>
                    </xsd:restriction>
                </xsd:simpleType>
            </xsd:attribute>

        </xsd:complexType>
    </xsd:element>

</xsd:schema>