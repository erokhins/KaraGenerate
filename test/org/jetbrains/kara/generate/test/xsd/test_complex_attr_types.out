AttributeTypeDecl:
   AttributeTypeDecl = (attrRestrictionEnumType, null, enumType)
   Values = [value1, value2]

   AttributeTypeDecl = (attrStrEnumType1, null, strEnumType)
   Values = [value1, value2]

   AttributeTypeDecl = (attrStrEnumType2, null, strEnumType)
   Values = [value-1, value-2]

   AttributeTypeDecl = (attrUnionEnumType, null, enumType)
   Values = [value1, value2]

AttributeDecl:
   AttributeDecl = (attrRestrictionEnumType, null, null})
   AttributeTypeDecl = (attrRestrictionEnumType, null)

   AttributeDecl = (attrStrEnumType1, null, null})
   AttributeTypeDecl = (attrStrEnumType1, null)

   AttributeDecl = (attrStrEnumType2, null, null})
   AttributeTypeDecl = (attrStrEnumType2, null)

   AttributeDecl = (attrUnionEnumType, null, null})
   AttributeTypeDecl = (attrUnionEnumType, null)

AttributeGroupDecl:
ElementDecl:
   ElementDeclaration = (complex, false)
   ElementGroups = []
   NewAllowElements = []
   AttributeGroups = []
   NewAttributes =
      AttributeDecl = (attrRestrictionEnumType, null, null})
      AttributeTypeDecl = (attrRestrictionEnumType, null)

      AttributeDecl = (attrStrEnumType1, null, null})
      AttributeTypeDecl = (attrStrEnumType1, null)

      AttributeDecl = (attrStrEnumType2, null, null})
      AttributeTypeDecl = (attrStrEnumType2, null)

      AttributeDecl = (attrUnionEnumType, null, null})
      AttributeTypeDecl = (attrUnionEnumType, null)


ElementGroup:
